### PR TITLE
monitor: trigger `MemorySet` trace event for `store_region`

### PR DESCRIPTION
**Pull request by SchmErik**
This PR fixes an issue where some memory operations done by ecalls do not show up in profiler outputs.

Fixes: #1194

### CODE REVIEW

**Comment by flaub** [comment]
Is this intended for 0.20?

**Comment by flaub** [comment]
The only issue I see with this is that the trace might get very large for an event for every byte in a large region. WDYT if we adjusted the `MemorySet` to take `addr` and `region` which would be a `Vec<u8>`? Maybe that's a bad idea...

**Comment by weikengchen** [comment]
Btw a separate solution is just to allow the profiler to access the memory if the profiler wants (passing an interface that allows profilers to access the memory), so that you do not have to provide the bytes together.

**Comment by SchmErik** [comment]
> Is this intended for 0.20?

Yeah that was my intention. My understanding of the rc- cycles is that once an rc is declared, we want to backport bug fixes but not any new features until 0.20 is cut but I think this was mostly centered around bonsai integration. I don't think we've talked about what should be done in this case where we fix a bug during the rc cycles specifically for the zkvm so I'm definitely open for further discussion/refinement of this process. @flaub do you have any thoughts on this?

**Comment by flaub** [comment]
I just need an indication that a PR is intended for a stable release, I think a way to do that is set the milestone.

**Comment by SchmErik** [comment]
> I just need an indication that a PR is intended for a stable release, I think a way to do that is set the milestone.

I see what you mean. I set the milestone on this PR about 2 weeks ago. I can see it on my end in this PR. Maybe there's an issue in github? Here's a screen shot:

Screenshot 2024-01-03 at 1 55 20 PM

File: risc0/zkvm/src/host/protos/api.proto [hunk 1]
```diff
@@ -239,7 +239,7 @@ message TraceEvent {
 
   message MemorySet {
     uint32 addr = 1;
-    uint32 value = 2;
+    bytes region = 2;
```
Comment (flaub) [inline risc0/zkvm/src/host/protos/api.proto:242]: This is a breaking change to protobuf. We can make breaking changes to the API, but we can't make breaking changes to protobuf messages. If we're changing the data type, we need to say:
```suggestion
    uint32 value = 2;
    bytes region = 3;
```

**Review by flaub — APPROVED** [review]

**Pull request merged**